### PR TITLE
Returndatacopy throws when offset+size is bigger than the size of returndata buffer

### PR DESCRIPTION
**Pull request by pirapira**
This follows the [change in the EIP](https://github.com/ethereum/EIPs/pull/211/commits/28f1709777cbd99dc96dcdbbef8c532674bfbc86#diff-fd8c393953fe66162d8bb518d87bfb57R43).

### CODE REVIEW

File: libevm/VM.cpp [hunk 1]
```diff
@@ -763,6 +763,9 @@ void VM::interpretCases()
 		{
 			if (!m_schedule->haveReturnData)
 				throwBadInstruction();
+			bigint const endOfAccess{bigint(m_SP[1]) + bigint(m_SP[2])};
```
Comment (chfast) [inline libevm/VM.cpp]: I would avoid explicit constructor here to still pretend bigint works like regular int.
```
bigint const endOfAccess = bigint(m_SP[1]) + bigint(m_SP[2]);
```

Comment (pirapira) [inline libevm/VM.cpp]: Let me pretend that way.

**Comment by pirapira** [comment]
@gumb0 @chriseth @chfast  ready to merge?

**Comment by chfast** [comment]
Yes.

**Pull request merged**